building/configuring_git_workflow.adoc

```diff
@@ -0,0 +1,426 @@
+
+To fetch sources, you must clone the repo to your local disk.
+
+[source,bash]
+----
+cd ~/my-github-stuff
+git clone git@github.com:jbosstools/jbosstools-base.git
+cd jbosstools-base
+----
+
+See also http://git-scm.com/book/en/v2/Git-Basics-Getting-a-Git-Repository#Cloning-an-Existing-Repository[Cloning an Existing Repository]
+
+Then, if you plan to contribute a pull request, you will need your own fork. To fork the repo, 
+browse to the repo on github, and click the `Fork` button in the top-right corner.
+
+image::../images/fork-a-repo.png[Fork a github repo]
+
+=== How to import projects into Eclipse
+
+Launch Eclipse, then select `File -> Import -> Maven -> Existing Maven Projects`
+
+image::../images/import-maven-project.png[]
+
+Browse for where you cloned the repo with which you want to work.
+
+image::../images/import-maven-project-2.png[]
+
+Some projects require a full Maven build before they will compile properly in Eclipse with PDE. To solve this problem, you can either build using Maven via the commandline, or configure Eclipse to use m2e and Tycho to build the project instead of PDE. The import wizard will do this for you automatically.
+
+image::../images/import-maven-project-3-tycho.png[]
+image::../images/import-maven-project-4-tycho.png[]
+
+Note that the Tycho configurator also includes PDE, so that it can be installed even if you're using a version of Eclipse which doesn't already include it.
+
+image::../images/import-maven-project-5-tycho-pde.png[]
+
+=== How to add forks to your local clone
+
+If you use https://hub.github.com/[hub], you can simply add a fork like this:
+
+[source,bash]
+----
+cd ~/my-github-stuff/jbosstools-base
+git remote add -p ${gituser}
+----
+
+Or, without https://hub.github.com/[hub], do this:
+
+[source,bash]
+----
+cd ~/my-github-stuff/jbosstools-base
+git remote add ${gituser} git@github.com:${gituser}/jbosstools-base.git
+----
+
+Then to check which forks you have attached:
+
+[source,bash]
+----
+git remove -v
+
+nickboldt   git@github.com:nickboldt/jbosstools-devdoc.git (fetch)
+nickboldt   git@github.com:nickboldt/jbosstools-devdoc.git (push)
+origin  git@github.com:jbosstools/jbosstools-devdoc.git (fetch)
+origin  git@github.com:jbosstools/jbosstools-devdoc.git (push)
+----
+
+
+== Submitting Pull Requests as a Contributor
+
+It is recommended that you use https://hub.github.com/[hub] to simplify your commandline git experience.
+
+To install hub, go here: https://hub.github.com/
+
+=== Pull Request Guidelines
+
+The rules are simple:
+
+. Ensure you've pulled the latest from the upstream origin's fork and topic branch
+. Rebase/merge as needed
+. Topic branch should contain JIRA ID
+. Commit comment(s) should also contain JIRA ID
+
+[source,bash]
+----
+# checkout the topic branch and origin branch, and link them
+# ${topic} = topic branch in your fork
+# ${branch} = branch in the origin
+git checkout origin/${branch} -b ${topic}
+git checkout ${topic}
```

Comment:
In the case of you already being on the right branch you just need to do "git checkout -b ${topic}" 
with yours you *must* use these two-commands - they cannot be separated.